pkg/management/component.go

```diff
@@ -302,12 +303,20 @@ func (c *component) CreateUser(ctx context.Context, realmName string, user api.U
 	var accessToken = ctx.Value(cs.CtContextAccessToken).(string)
 	var ctxRealm = ctx.Value(cs.CtContextRealm).(string)
 
-	var userRep kc.UserRepresentation
-
-	userRep = api.ConvertToKCUser(user)
+	var userRep = api.ConvertToKCUser(user)
 
-	// Store user in KC
-	locationURL, err := c.keycloakClient.CreateUser(accessToken, ctxRealm, realmName, userRep)
+	var locationURL string
+	var err error
+	if realmConf, errConf := c.configDBModule.GetAdminConfiguration(ctx, realmName); errConf != nil {
+		err = errConf
+	} else if realmConf.SelfRegisterEnabled != nil && *realmConf.SelfRegisterEnabled {
```

Comment:
I don't understand why we have such condition.
We can have selfregister in corporate.
I think the condition should a test on realmName == socialRealm ?

Reply:
Done